internal/clients/diffreviewer/circleci/circleci_service.go

```diff
@@ -259,3 +270,33 @@ func (s *Service) createGithubRepositoryComments(comments []*diffreviewer.Commen
 	}
 	return githubComments
 }
+
+type prComment struct {
+	Body string
+	Path string
+	Line int
+}
+
+func filterExistingComments(comments []*github.PullRequestComment, existingComments []*github.PullRequestComment) []*github.PullRequestComment {
+	existingCommentsMap := make(map[prComment]bool, len(existingComments))
+	for _, ec := range existingComments {
+		comment := prComment{
+			Body: *ec.Body,
+			Path: *ec.Path,
+			Line: *ec.Line,
+		}
+		existingCommentsMap[comment] = true
+	}
+	filteredComments := []*github.PullRequestComment{}
```

Comment:
i imagine the capacity for filteredComments will typically be around the length of `comments` (since hopefully users will fix their sensitive info more often than not), so I think we can specify a slice capacity here